subsys/net/ip/tcp.c

```diff
@@ -1879,48 +1789,46 @@ static void tcp_resend_data(struct k_work *work)
 		goto out;
 	}
 
-	if (IS_ENABLED(CONFIG_NET_TCP_CONGESTION_AVOIDANCE) &&
-	    (conn->send_data_retries == 0)) {
-		tcp_ca_timeout(conn);
-		if (tcp_window_full(conn)) {
-			(void)k_sem_take(&conn->tx_sem, K_NO_WAIT);
-		}
-	}
-
-	conn->data_mode = TCP_DATA_MODE_RESEND;
-	conn->unacked_len = 0;
-
-	ret = tcp_send_data(conn);
-	conn->send_data_retries++;
-	if (ret == 0) {
-		if (conn->in_close && conn->send_data_total == 0) {
-			NET_DBG("TCP connection in %s close, "
-				"not disposing yet (waiting %dms)",
-				"active", tcp_max_timeout_ms);
-			k_work_reschedule_for_queue(&tcp_work_q,
-						    &conn->fin_timer,
-						    FIN_TIMEOUT);
-
-			conn_state(conn, TCP_FIN_WAIT_1);
-
-			ret = tcp_out_ext(conn, FIN | ACK, NULL,
-					  conn->seq + conn->unacked_len);
-			if (ret == 0) {
-				conn_seq(conn, + 1);
```

Comment:
I see those lines were completely dropped - is my thinking correct that this action is simply not needed here, as transmitting FIN is already hanlded in ESTABLISHED state in `tcp_in()`, when ACK is received
https://github.com/shrek-wang/zephyr/blob/c5b847851b23024f0b3ec0aade4250052cc32167/subsys/net/ip/tcp.c#L3147
?